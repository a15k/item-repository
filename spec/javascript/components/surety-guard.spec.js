import renderer from 'react-test-renderer';
import SuretyGuard from 'components/surety-guard';


describe(SuretyGuard, () => {
  let props;

  beforeEach(() => {
    props = {
      header:    'Sure are you?',
      message:   'Yo! are you sure?',
      placement: 'bottom',
      onConfirm: jest.fn(),
    };
  });

  it('renders', () => {
    const guard = renderer.create(
      <SuretyGuard {...props}><p>Hi</p></SuretyGuard>
    );
    expect(guard.toJSON()).toMatchSnapshot();
    guard.unmount();
  });

  // code below works but throws promise rejection warnings
  // popper can't be mocked? see Apr 5 comment on https://github.com/FezVrasta/popper.js/issues/478
  xit('renders into dom', () => {
    const guard = mount(<SuretyGuard {...props}><p>Hi</p></SuretyGuard>);
    guard.find('span').simulate('click');
    expect(
      document.body.querySelector('.popover-inner').innerHTML
    ).toMatchSnapshot();
  });

});
