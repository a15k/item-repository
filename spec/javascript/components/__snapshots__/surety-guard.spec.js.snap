// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuretyGuard renders 1`] = `
<span
  className="surety-guard"
  onClick={[Function]}
  role="button"
  tabIndex="0"
>
  <p>
    Hi
  </p>
</span>
`;

exports[`SuretyGuard renders into dom 1`] = `"<h3 class=\\"popover-header\\">Sure are you?</h3><div class=\\"popover-body\\">Yo! are you sure?</div><div class=\\"sc-bdVaJa jVFQRp\\"><button type=\\"button\\" class=\\"btn btn-secondary\\">Cancel</button><button type=\\"button\\" class=\\"btn btn-primary\\">OK</button></div>"`;
